### PR TITLE
InstructorSummary: Instructors should see assigned TAs

### PR DESCRIPTION
**Pull request by ltwheeler**
Part of this work ensures that instructors will see TAs assigned to their course, but only if those assignments have been 'opened for review'.

### CODE REVIEW

File: app/summary/services/summaryStateService.js [hunk 1]
```diff
@@ -349,6 +349,50 @@ summaryApp.service('summaryStateService', function ($rootScope, $log, Course, Sc
...
+							supportStaffList.list[staff.id] = staff;
+							supportStaffList.ids.push(staff.id);
+						});
+
+						// Add TA data to sectionGroups
+						slotSectionGroup.teachingAssistants = [];
+
+						action.payload.supportAssignments.forEach(function(supportAssignment) {
+							if (supportAssignment.appointmentType == "teachingAssistant" && supportAssignment.sectionGroupId == sectionGroup.id) {
+
```
Comment (cthielen) [inline app/summary/services/summaryStateService.js]: Unnecessary blank line

File: app/summary/services/summaryStateService.js [hunk 2]
```diff
@@ -349,6 +349,50 @@ summaryApp.service('summaryStateService', function ($rootScope, $log, Course, Sc
...
+								var supportStaff = supportStaffList.list[supportAssignment.supportStaffId];
+
+								slotSectionGroup.teachingAssistants.push({
```
Comment (cthielen) [inline app/summary/services/summaryStateService.js]: Re-do this forEach loop + if statement as: slotSectionGroup.teachingAssistants = action.payload.supportAssignments.filter(the_if_conditions).map(the_push_contents). Cleaner, right?

Comment (ltwheeler) [reply app/summary/services/summaryStateService.js]: I would agree, looks like a straightforward candidate for filter.

**Comment by cthielen** [comment]
How much work would it be to add the following functionality:

1) If a course has no TA slots, the instructor summary should list "Will not use TAs"
2) If a course has TA slots but no assignments, list "No TAs assigned"

Also, the "open for instructor review" -- any idea who requested that?

**Comment by ltwheeler** [comment]
I don't recall the specific workgroup but I know it was a few that mentioned it.

Controlling information regarding TA assignments is a touchy issue in this process. Departments have gotten into trouble for various forms of 'implying' or 'suggesting' someone will get a TAship who then didn't.

Regarding strictly controlling 'instructor review', basically some workgroups said instructors could often not be trusted to keep draft copies to themselves, so they would try to limit instructor access as well.

It does feel 'not great' to me that they have to remember to flip that flag in order for grad students to show up, and my guess that the workgroups wanting strict control on instructors are probably in the minority.

**Comment by cthielen** [comment]
The TA viewership makes sense to me, the faculty part less so. I understand the draft copy issue but this would be a faculty logging into IPA and revealing it to a grad student. Do you think it's worth it if I reach out to the customers and see if they'll accept not having the faculty review flag, i.e. always open? We could preference text with "(draft)" to add additional legal help.

**Comment by ltwheeler** [comment]
It wouldn't be hard to restructure things to support those two messages you described,

But when do we want to display that to the instructor? 'Will not use TAs' sounds like a decision, whereas it could be the result of simply not making the TA slots yet.

The teaching call is designed to update for the instructors gradually over time as the draft comes into focus, but the support call currently is not.

We should probably either be more explicit and say something like 'TA planning is still underway.' until the open review flag is flipped,

or investigate if just letting instructors see the drafts as they change would maybe be fine. Or maybe we could change it to be 'opt-out' instead of opt-in? A new schedule would then give instructors access, and block grad students?

**Comment by ltwheeler** [comment]
I think its worth reaching out to see if they'll accept the instructor review flag going away.

I recall a story from one of them where an instructor did exactly what you described, got a copy of the draft and showed their grad students to help them plan their quarter or something.

And maybe a mix of the two would be sensible:
Let instructor always see's the current 'draft' data for TAs in their courses, but control access to the 'support call review' that shows everything, via the flag.

**Pull request merged**